docs/source/best-practices/checkpointing.rst

```diff
@@ -110,45 +167,67 @@ Where the checkpointing implementation is given in this code snippet of ``train_
+The entire scripts can be found `here <https://github.com/northeastern-rc/training-checkpointing/tree/main/Exercise_2>`_ and were modified from `TensorFlow Save and load models <https://www.tensorflow.org/tutorials/keras/save_and_load>`_.
 
-The Slurm option ``--array=1-10%1`` will create 10 Slurm array tasks, and will run one task job at a time. Note that the saved variable ``%A`` denotes the main job ID, while variable ``%a`` denotes the task ID (spanning values 1-10). Note that also the output/error files are unique in order to prevent different jobs writing to the same files.
-The Shell variable ``SLURM_ARRAY_TASK_ID`` holds the unique task ID value and can be used within the Slurm Shell script to point to different files or variables.
+The Slurm option, ``--array=1-10%1``, will create 10 Slurm array tasks and run one task at a time. Note that the saved variable ``%A`` denotes the main job ID, while variable ``%a`` indicates the task ID (spanning values 1-10). Note that the output/error files are also unique to prevent different jobs from writing to the same files.
+The Shell variable, ``SLURM_ARRAY_TASK_ID``, holds the unique task ID value and can be used within the Slurm Shell script to point to different files or variables.
 
 To submit this job to the scheduler, use the command::
-   
-  sbatch submit_tf_array.bash
-
-Checkpointing using DMTCP
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-`DMTCP <https://dmtcp.sourceforge.io/>`_ (Distributed MultiThreaded checkpointing) is a checkpointing tool that lets you checkpoint without the need to change your code. It Works with most Linux applications such as Python, Matlab, R, GUI, MPI etc. 
-The program runs in the background of your program, without significant performance loss, and saves the process states into checkpoint files. DMTCP is available on the cluster ::
-
- module avail dmtcp
- module show dmtcp
- module load dmtcp/2.6.0
-
-As DMTCP runs in the background, it requires some changes to your Shell script. For examples of how to checkpoint with DMTCP visit `here <https://github.com/northeastern-rc/training-checkpointing/tree/main/Exercise_3>`_. 
-The example demonstrates how to use DMTCP with a simple C++ program (scripts modified from `RSE-Cambridge <https://github.com/RSE-Cambridge/dmtcp-tests>`_).
-
-
-Checkpointing tips
-~~~~~~~~~~~~~~~~~~~
 
-What data to save?
- * Non-temporary application data
- * Any application data that has been modified since the last checkpoint
- * Delete checkpoints that are no longer useful - keep only the most recent checkpoint file.
+  sbatch submit_tf_array.bash
 
-How frequently to checkpoint? 
- * Too often – will slow down your calculation, may be I/O heavy and memory-limited.
- * Too infrequently – leads to large/long rollback times.
- * Consider how long it takes to checkpoint and restart your calculation. 
- * In most cases a rate of every 10-15 minutes is ok.
+Python PyTorch
```

Comment:
Is this heading a placeholder or can we change to Python Pytorch tips and tricks?